pkg/controller/ingress-shim/controller_test.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	testpkg "github.com/jetstack/cert-manager/pkg/controller/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kclient "k8s.io/client-go/kubernetes"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+var _ = cmapi.Certificate{}
+
+func Test_controller_Register(t *testing.T) {
+	tests := []struct {
+		name              string
+		existingKObjects  []runtime.Object
+		existingCMObjects []runtime.Object
+		givenCall         func(*testing.T, cmclient.Interface, kclient.Interface)
+		expectRequeueKey  string
+	}{
+		{
+			name: "ingress should be queued when it is created",
```

Comment:
These scenario- specific test case names are really helpful for understanding how this works 👍🏼